fhirstarter/exceptions.py

```diff
@@ -104,7 +104,7 @@ def __init__(self) -> None:
 
     def operation_outcome(self) -> OperationOutcome:
         try:
-            _, resource_type_str, id_ = self._request.url.components.path.split("/")  # type: ignore
+            resource_type_str, id_ = self._request.url.components.path.split("/")[-2:]
```

Comment:
From the initial source code, and the faced issue, I must admit, I definitively prefer my code proposal.
It's a very tiny change, definitively upgrade ascending safe, knowing it's just putting to trash the "_" variables, which wasn't used afterward in the function...

Nevertheless, if you propose a complete code, aligned with your approach, count on me for testing.

Regards.

Reply:
I did some reading on mounted sub-applications to learn more about them (I haven't used them before). After reading that, I can see what your concern was. I think it is correct to essentially parse the path from tail to head instead of head to tail.

I do, however, feel strongly that request parsing should live in a central place, for consistency and testing purposes. The helper function I mentioned also contains the flaw that you identified. I know that function is used by FHIR implementations, so it's not easy for me to go for a narrow fix here and leave that flaw in place. Hardening and centralizing request parsing resolves this concern.

I submitted a draft PR that I believe would address the issue you've identified and resolves my own concerns here: https://github.com/canvas-medical/fhirstarter/pull/85

Reply:
Hi,
Thanks for this correction,
I do confirm your pull request #85 is perfectly & successfully correcting the issue I was previously facing.
Great thanks.